src/components/SideBar/SideBar.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import "./SideBar.css";
+import { Link } from "react-router-dom";
+import { AuthContext } from "../../contexts/spotifyContext";
+
+function SideBar() {
+  //const [accessToken, setAccessToken] = useState(null);
+  const [userData, setUserData] = useState({
+    display_name: "",
+    images: [],
+  });
+  const accessToken = localStorage.getItem("spotify_access_token");
+  useEffect(() => {
+    // const accessToken = localStorage.getItem("spotify_access_token");
+    console.log(accessToken);
+    if (accessToken) {
+      fetch("https://api.spotify.com/v1/me", {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          setUserData({
+            display_name: data.display_name,
+            images: data.images,
+          });
+        })
+        .catch((err) => console.error("Error fetching user data:", err));
+    }
+  }, [accessToken]);
+
+  return (
+    <div className="sidebar">
+      <div className="sidebar__user">
+        <img
+          src={userData.images[0]?.url}
+          alt="user image"
+          className="sidebar__user_avatar"
+        ></img>
+        <p className="sidebar__user_username">{userData.display_name}</p>
```

Comment:
`sidebar__user_avatar`
`sidebar__user_username`
and so on


**Modifier** should be used only with its modifiable **basic** class: `block__element block__element_modifier-key_modifier-value`. Elements should get only **modifying styles** from modifier, and all other styles should be got from the basic class.


But if it's not a modifier, then you should not use `_` in it.

`sidebar__user-avatar`
`sidebar__user-username`

**Please, correct it everywhere in the code**